[Ingyin] iP

broccoli-hater opened the pull request:
# SirTalksALot

> SirTalksALot provides an entertaining way to keep track of your tasks,
> **Warning:** It does talk a lot.

   - text-based
   - easy to learn
   - (hopefully) entertaining to interact with

All you need to do is,

1. download it from [here](https://github.com/broccoli-hater/ip/releases/tag/A-Jar/).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you

And it is **FREE**!

Features:

- [x] Managing tasks
- [x] Managing deadlines 
- [x] Managing events
- [ ] Reminders (coming soon)

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

broccoli-hater committed:
rename, greet, exit

broccoli-hater committed:
added personality and echo

broccoli-hater committed:
added add, list

broccoli-hater committed:
added mark as done

broccoli-hater committed:
added todo, event and deadline

broccoli-hater committed:
added automated text ui testing

broccoli-hater committed:
added error handling and new input test cases

broccoli-hater committed:
added delete and new input test cases

broccoli-hater committed:
Fix some code standardization

broccoli-hater committed:
Fix code structure

broccoli-hater committed:
Add data saving

broccoli-hater committed:
Add load data

broccoli-hater committed:
Fix file path OS dependency

broccoli-hater committed:
Merge branch 'branch-Level-7'

broccoli-hater committed:
Add dates

broccoli-hater committed:
Merge branch 'branch-Level-8'

broccoli-hater committed:
Add Storage and Ui classes

broccoli-hater committed:
Update Ui class

broccoli-hater committed:
Add Tasklist

broccoli-hater committed:
Add Parser and Command classes

broccoli-hater committed:
Merge branch 'branch-OOP'

broccoli-hater committed:
Separate Command subclasses

broccoli-hater committed:
Add packages

broccoli-hater committed:
Merge branch 'branch-packages'

broccoli-hater committed:
Merge branch 'add-gradle-support'

broccoli-hater committed:
Add gradle support and javadoc comments

broccoli-hater committed:
Add JUnit tests

broccoli-hater committed:
Add JUnit tests

broccoli-hater committed:
Add JUnit tests

broccoli-hater committed:
Create a JAR file

broccoli-hater committed:
Add find function

broccoli-hater committed:
Fix coding standards

broccoli-hater committed:
Merge branch 'branch-Level-9'

cyhni commented on src/main/java/util/Ui.java:
```diff
@@ -0,0 +1,172 @@
+    /**
+     * Prints a line break to separate sections of the conversation.
+     */
+    public static void breakLine() {
```
Maybe `printBreakLine()` might be a clearer method name for this.

broccoli-hater replied:
Noted! Thanks for catching this issue.

cyhni commented on src/main/java/util/Ui.java:
```diff
@@ -0,0 +1,172 @@
+    /**
+     * Informs the user that the specified task does not exist.
+     */
+    public static void taskNotFound() {
```
It would be good if the name for this method could be made to sound more like an active verb. For example, maybe have it named `printTaskNotFound()`
I found similar issues in several other places in the code as well.
Do take note!

cyhni commented on src/main/java/util/Storage.java:
```diff
@@ -0,0 +1,119 @@
+            if (f.exists()) {
+                Scanner sc = new Scanner(f);
+                while (sc.hasNextLine()) {
+                    String[] task = sc.nextLine().split(" ");
```
Consider making String array `task` to be in plural form to be clearer.

cyhni commented on src/main/java/util/Parser.java:
I like how you structured your code here! Good job! 👍

cyhni commented on src/test/java/task/TaskListTest.java:
```diff
@@ -0,0 +1,50 @@
+ */
+public class TaskListTest {
+    @Test
+    public void testToString() {
```
Maybe you could standardise your test names with your other tests. 
Test naming conventions are thoroughly run through [HERE](https://se-education.org/se-book/cppToJava/junit/basic/index.html)

cyhni reviewed (commented):
Overall your code quality on naming is excellent except for some method names in the `Ui` class. 
Very clean structure of code that made it really easy to read and review.
Keep up the good work! 🎉

broccoli-hater committed:
Add basic GUI

broccoli-hater committed:
Add GUI responses

broccoli-hater committed:
Debug color change for different commands

broccoli-hater committed:
Merge branch 'branch-GUI'

broccoli-hater committed:
Add assertions

broccoli-hater committed:
Merge branch 'branch-A-Assertations'

broccoli-hater committed:
Storage class: refactor methods

The loadData() method is deeply nested and can be refactored for better readability.
The switch case inside the method have some common behaviours.

Let's,
* Extract the common behavior into a separate method to remove code duplication
* Simplify the parsing logic and improve variable naming for better readability

broccoli-hater committed:
SirTalksALot class: remove redundant code
Ui class: remove redundant code

Dead code should be removed because it makes the code harder to read or may contain bugs

Let's remove redundant methods, imports and comments

broccoli-hater committed:
Refactor Parser and extract duplicate date parsing code into a method

broccoli-hater committed:
Merge branch 'master' into branch-A-CodeQuality

broccoli-hater committed:
Merge pull request #1 from broccoli-hater/branch-A-CodeQuality

Branch a code quality

broccoli-hater committed:
Merge branch 'master' of https://github.com/broccoli-hater/ip

broccoli-hater committed:
Add tag system

broccoli-hater committed:
Add Tag Command for adding new tags to a task in the task list

broccoli-hater committed:
Add Untag Command to remove tags from a task in the task list

broccoli-hater committed:
Improve file formatting in Storage Class

broccoli-hater committed:
Merge branch 'branch-extension'

broccoli-hater committed:
Fix bug in parsing of ToDo task

broccoli-hater committed:
Add more error handling for parsing tag and untag commands

broccoli-hater committed:
Add user guide in readme file

broccoli-hater committed:
Fix minor formatting in readme file

broccoli-hater committed:
Add javadoc comments

broccoli-hater committed:
Add Junit testing for TaskListTest.java

broccoli-hater committed:
Add Junit tests for TagCommandTest and ToDoTest

broccoli-hater committed:
Fix checkstyle errors
